File: src/main/scala/org/raisercostin/jedi/impl/LogTry.scala
package org.raisercostin.jedi.impl

import scala.util.{ Try, Success, Failure }

object LogTry {
  implicit class LogTry[A](res: Try[A]) extends SlfLogger{
    def log() = res match {
      case Success(s) =>
        res
      case Failure(f) =>
        logger.info("Failure via LogTry.",f) 
        res
    }
  }
}

File: src/main/kotlin/org/raisercostin/jedi/impl/LogTry.kt
package org.raisercostin.jedi.impl

import scala.util.Try
import scala.util.Success
import scala.util.Failure

object LogTry {
  implicit class LogTry<+A>(res: Try<A>) : SlfLogger{
    fun log():Try<A> = res when {
      Success(s) ->
        res
      Failure(f) ->
        logger.info("Failure via LogTry.",f)
        res
    }
  }
}